From b8390499d8c6a020eac3720d44bef5f663b337cc Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@st.com>
Date: Wed, 23 Nov 2022 12:43:40 +0100
Subject: [PATCH 096/529] ARM: config: Enable STM32 cpuidle

Allow cpu idle configuration.
This will add menu governor on STM32MP SoCs

Signed-off-by: Pascal Paillet <p.paillet@st.com>
Change-Id: I4bf35f4098ec1b0c5df104bc61e15437978c11bb
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/226987
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Alexandre TORGUE <alexandre.torgue@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/276997
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Domain-Review: Alexandre TORGUE <alexandre.torgue@foss.st.com>
---
 arch/arm/configs/fragment-01-multiv7_cleanup.config | 1 -
 arch/arm/configs/fragment-02-multiv7_addons.config  | 3 +++
 2 files changed, 3 insertions(+), 1 deletion(-)

--- a/arch/arm/configs/fragment-01-multiv7_cleanup.config
+++ b/arch/arm/configs/fragment-01-multiv7_cleanup.config
@@ -47,7 +47,6 @@
 # CONFIG_CAN_RCAR is not set
 # CONFIG_CHROME_PLATFORMS is not set
 # CONFIG_CPU_FREQ is not set
-# CONFIG_CPU_IDLE is not set
 # CONFIG_DETECT_HUNG_TASK is not set
 # CONFIG_EFI is not set
 # CONFIG_ENABLE_MUST_CHECK is not set
--- a/arch/arm/configs/fragment-02-multiv7_addons.config
+++ b/arch/arm/configs/fragment-02-multiv7_addons.config
@@ -1,9 +1,12 @@
 #  ALPHABETICALLY SORTED: $LC_ALL=C sort fragment-02-multiv7_addons.config -o fragment-02-multiv7_addons.config
 CONFIG_AEABI=y
+CONFIG_ARM_CPUIDLE=n
 CONFIG_ARM_SCMI_PROTOCOL=y
 CONFIG_ARM_SMC_MBOX=y
+CONFIG_ARM_STM32_CPUIDLE=y
 CONFIG_CHR_DEV_SG=y
 CONFIG_COMMON_CLK_SCMI=y
+CONFIG_CPU_IDLE=y
 CONFIG_CRC_ITU_T=m
 CONFIG_DEBUG_INFO=y
 CONFIG_DEBUG_INFO_DWARF_TOOLCHAIN_DEFAULT=y
